tests/accuracy_tests/sim_for_alphapeel_accu_test/sim_for_metafounder_accu.R

```diff
@@ -0,0 +1,267 @@
+rm(list = ls())
+# install.packages(pkg = "AlphaSimR")
+library(AlphaSimR)
+
+# ---- Two (or more) sub-populations A and B which cross in later generations ----
+
+parameters <- read.table("../simulation_parameters.txt")
+nparams <- nrow(parameters)
+for (parameter in (1:nparams)) {
+  eval(parse(text = paste0(parameters$V1[parameter], "<-", parameters$V2[parameter])))
+}
+
+nIndPerGen <- nInd / nGen
+
+nLociAllPerChr <- floor(nLociAll / nChr)
+nLociHDPerChr <- nLociHD / nChr
+nLociLDPerChr <- nLociLD / nChr
+
+founderGenomes <- runMacs(nInd = nIndPerGen, nChr = nChr, segSites = nLociAllPerChr, split = 1000)
+SP <- SimParam$new(founderPop = founderGenomes)
+SP$setSexes("yes_rand")
+SP$setTrackPed(isTrackPed = TRUE)
+SP$setTrackRec(isTrackRec = TRUE)
+varG <- matrix(data = c( 1.0, -0.3,
+                         -0.3,  1.0), byrow = TRUE, nrow = 2)
+SP$addTraitA(nQtlPerChr = 1000, mean = c(0, 0), var = diag(varG), corA = varG)
+varE <- matrix(data = c(2.0, 0.0,
+                        0.0, 2.0), byrow = TRUE, nrow = 2)
+
+collectData <- function(pop, data = NULL, population, generation) {
+  remove <- FALSE
+  if (is.null(data)) {
+    remove <- TRUE
+    data <- vector(mode = "list", length = 5)
+    names(data) <- c("pedigree", "haploIBD", "genoIBS", "Haplotype", "Genotype")
+    data$pedigree <- data.frame(id = NA, population = NA, generation = NA,
+                                mid = NA, fid = NA,
+                                gv1 = NA, pv1 = NA,
+                                gv2 = NA, pv2 = NA)
+    data$haploIBD <- matrix(data = NA, ncol = sum(pop@nLoci))
+    data$genoIBS <- matrix(data = NA, ncol = sum(pop@nLoci))
+    data$Haplotype <- matrix(data = NA, ncol = sum(pop@nLoci))
+    data$Genotype <- matrix(data = NA, ncol = sum(pop@nLoci))
+  }
+  data$pedigree <- rbind(data$pedigree,
+                         data.frame(id = pop@id,
+                                    population = population,
+                                    generation = generation,
+                                    mid = pop@mother,
+                                    fid = pop@father,
+                                    gv1 = pop@gv[, 1],
+                                    pv1 = pop@pheno[, 1],
+                                    gv2 = pop@gv[, 2],
+                                    pv2 = pop@pheno[, 2]))
+  data$haploIBD <- rbind(data$haploIBD,
+                         pullIbdHaplo(pop = pop))
+  data$genoIBS <- rbind(data$genoIBS,
+                        pullSegSiteGeno(pop = pop))
+  listLoci <- seq.int(from = 1, to = sum(pop@nLoci), by = 1)
```

Comment:
@RosCraddock do you need this listLoci bit of code? Also, isn't your Genotype the same as genoIBS? I got the later via pullSegSiteGeno() and there is also pullSegSiteHaplo() which will give you haploIBS. That's all you need I think,